pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -765,64 +765,98 @@ func TestGetTaskFunc_VerifySuccess(t *testing.T) {
 	resolvedMatched := test.NewResolvedResource(signedTaskBytes, nil, matchPolicyRefSource, nil)
 	requesterMatched := test.NewRequester(resolvedMatched, nil)
 
+	warnPolicyRefSource := &v1beta1.RefSource{
+		URI: "	warnVP",
+	}
+	resolvedUnsignedMatched := test.NewResolvedResource(unsignedTaskBytes, nil, warnPolicyRefSource, nil)
+	requesterUnsignedMatched := test.NewRequester(resolvedUnsignedMatched, nil)
+
 	taskRef := &v1beta1.TaskRef{ResolverRef: v1beta1.ResolverRef{Resolver: "git"}}
 
+	// the test covers those cases:
+	// VerificationPass: 1) 2) 3) signed task passes when no-match-policy is set to fail, warn, ignore
+	// VerificationWarn: 4) 5) 6) warn no-match-policy and failed warn policy
+	// VerificationSkip: 7) 8) ignore no match policies
 	testcases := []struct {
-		name                      string
-		requester                 *test.Requester
-		verificationNoMatchPolicy string
-		policies                  []*v1alpha1.VerificationPolicy
-		expected                  runtime.Object
-		expectedRefSource         *v1beta1.RefSource
+		name                       string
+		requester                  *test.Requester
+		verificationNoMatchPolicy  string
+		policies                   []*v1alpha1.VerificationPolicy
+		expected                   runtime.Object
+		expectedRefSource          *v1beta1.RefSource
+		expectedVerificationResult *trustedresources.VerificationResult
 	}{{
-		name:                      "signed task with matching policy pass verification with enforce no match policy",
-		requester:                 requesterMatched,
-		verificationNoMatchPolicy: config.FailNoMatchPolicy,
-		policies:                  vps,
-		expected:                  signedTask,
-		expectedRefSource:         matchPolicyRefSource,
+		name:                       "warn unsigned task fails matching policies",
```

Comment:
I am having trouble telling if there is a matching policy here. I'm guessing there is because we have `FailNoMatchPolicy`, but the VerificationResult is a warning, but I'm still a bit confused.

Reply:
Yes, this test case is used to test when a task fails `warn` mode policy, the verification result should be warn type. Maybe I should rename the test name to "task fails warn mode policy return warn verificationresult"